mne/decoding/time_gen.py

```diff
@@ -204,16 +229,32 @@ def predict(self, epochs):
         if not hasattr(self, 'estimators_'):
             raise RuntimeError('Please fit models before trying to predict')
 
-        # clean attributes
+        # Check predict mode
+        if self.predict_mode not in ['cross-validation', 'mean-prediction']:
+            raise ValueError('`predict_mode` must be a str, "mean-prediction" '
+                             'or "cross-validation"')
+
+        # Check that training cv and predicting cv match
+        if self.predict_mode == 'cross-validation':
+            n_est_cv = [len(estimator) for estimator in self.estimators_]
+            if hasattr(self.cv_, 'get_n_splits'):
+                n_split = self.cv_.get_n_splits()
+            else:
+                n_split = self.cv_.n
+            if ((len(set(n_est_cv)) != 1) or (n_est_cv[0] != n_split) or
+                    len(self.y_train_) != len(epochs)):
+                raise ValueError(
+                    'When `predict_mode = "cross-validation"`, the training '
+                    'and predicting cv schemes must be identical.')
+
+        # Clean attributes
```

Comment:
no, it's a bit faster. The script used to check these inputs inside the parallel loops (hence leading to redundant computations). Now we just check everything before starting joblib. 

The main advantage is for debugging: when it crashes in parallel, it takes a while before the error is thrown back, and you can't enter debugging mode.